packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1150,6 +1150,18 @@ mixin TextInputClient {
   /// [TextInputClient] should cleanup its connection and finalize editing.
   void connectionClosed();
 
+  /// The framework calls this method to notify that the text input control has
+  /// been changed.
+  ///
+  /// The [TextInputClient] may switch to the new text input control by hiding
+  /// the old and showing the new input control.
+  ///
+  /// See also:
+  ///
+  ///  * [TextInputControl.hide], a method to hide the old input control.
+  ///  * [TextInputControl.show], a method to show the new input control.
+  void didChangeInputControl(TextInputControl? oldControl, TextInputControl? newControl);
```

Comment:
I think you should add an empty implementation here, unless you disagree.  As it is, this will be a breaking change for anyone with a custom TextInputClient.  If you add an empty implementation, then users using `with` with a custom TextInputClient won't be broken while those using `implements` will (their choice).

My guess is that users with apps that don't care about virtual keyboards won't care about this and won't want to be broken, but users with custom text editors etc. may want to be made aware of this so they can support it.

See this [design doc](https://docs.google.com/document/d/1OxDsf_eot7TlsRn57z-1_dGbvuqnMxO7QEGIAmOCZtA/edit) and a [dartpad](https://dartpad.dartlang.org/?id=112a32907c0d904cfeb84dbca0dadd44) example.

Reply:
Agreed :+1: